src/modules/kvs/kvs.c

```diff
@@ -1223,6 +1243,47 @@ static int heartbeat_root_cb (struct kvsroot *root, void *arg)
     return 0;
 }
 
+static void sync_continuation (flux_future_t *f, void *arg)
+{
+    struct kvs_ctx *ctx = arg;
+
+    if (flux_rpc_get (f, NULL) < 0)
+        flux_log_error (ctx->h, "%s: flux_rpc_get", __FUNCTION__);
+    flux_future_destroy (f);
+    ctx->sync_f = NULL;
+    ctx->sync_started = false;
```

Comment:
Should this be set false in `sync_cb()`?  If another commit completes before an in-progress sync does, then it wouldn't start the timer again, and could sit there until another commit comes along.

Reply:
I don't think so.  The timer could be restarted before the last sync completes.  Here's the way I think about it.  Here's the transaction queue where a bunch of commits come in and are put on the queue.

`putA -> putB -> putC -> NULL`

`putA` finishes, leading to the sync timer to be started.  At some point in the future the `sync_cb()` will be called and add a transaction to the end of the queue to do the sync.  So perhaps its looks like this at that point (`putB` has completed before the `sync` was added):

`putC -> putD -> putE -> sync -> NULL`

if we set `sync_started` to false in `sync_cb()`, that means the timer can be restarted after `putC` completes, which is not what we want.  We'd want the sync timer to be restarted on a `putF` that will come later after the sync.

But your comment does make me think, is it possible a `putF` could complete before the `sync_started` flag is set to false in the continuation?  The odds are low, but I'm not sure if it could be guaranteed.  There's a lot of reactor stuff going on in the background.

I suppose the safe thing is `sync_started` would maybe have to be set to false when the sync is simply completed, perhaps via some callback mechanism.

These corner cases are tough to think about wrap around, and I definitely had to ponder them a bit.  Is it worth considering going back to something closer to the initial implementation I proposed?  Where its simply a timer that sets off every X time?  It largely avoids thinking about a lot of these corner cases.